### PR TITLE
Redisdead v1.9

### PR DESCRIPTION
**Pull request by regit**
Update of #1522:
- Don't use atomic because we have a lock
- Merge some commits
- Rebase on latest master

PR scripts:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/80
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/78

### CODE REVIEW

File: src/util-logopenfile.c [hunk 1]
```diff
@@ -311,3 +462,108 @@ int LogFileFreeCtx(LogFileCtx *lf_ctx)
...
+        }
+    } else {
+        redisReply *reply = redisCommand(file_ctx->redis, "%s %s %s",
+                file_ctx->redis_setup.command,
+                file_ctx->redis_setup.key,
+                string);
+
+        switch (reply->type) {
+            case REDIS_REPLY_ERROR:
+                SCLogWarning(SC_WARN_NO_UNITTESTS, "Redis error: %s", reply->str);
```
Comment (inliniac) [inline src/util-logopenfile.c]: Strange error code, is that intentional?

Comment (regit) [inline src/util-logopenfile.c]: Humm, it looks like this code is doomed.

**Comment by regit** [comment]
@inliniac I let you review if you find any other stupidity. Resubmitting once it is done.

File: suricata.yaml.in [hunk 2]
```diff
@@ -44,6 +44,10 @@ host-mode: auto
 #  user: suri
 #  group: suri
 
+# Some logging module will use that name in event as identifier. The default
+# value is the hostname
+#sensor-name: suricata
```
Comment (inliniac) [inline suricata.yaml.in]: Wonder if we should use this for all json output instead of just redis

**Comment by inliniac** [comment]
Done @regit :)

**Pull request closed without merging**